engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala

```diff
@@ -19,40 +21,44 @@ object ProcessState {
   implicit val uriEncoder: Encoder[URI] = Encoder.encodeString.contramap(_.toString)
   implicit val uriDecoder: Decoder[URI] = Decoder.decodeString.map(URI.create)
 
-  def apply(deploymentId: String, status: StateStatus, version: Option[ProcessVersion], allowedActions: List[ProcessActionType]): ProcessState =
-    ProcessState(DeploymentId(deploymentId), status, version, allowedActions)
+  def apply(deploymentId: String, status: StateStatus, version: Option[ProcessVersion], definitionManager: ProcessStateDefinitionManager): ProcessState =
+    ProcessState(DeploymentId(deploymentId), status, version, definitionManager, Option.empty, Option.empty, List.empty)
 
   def apply(deploymentId: DeploymentId,
             status: StateStatus,
             version: Option[ProcessVersion],
             definitionManager: ProcessStateDefinitionManager,
             startTime: Option[Long],
             attributes: Option[Json],
-            errorMessage: Option[String]): ProcessState =
+            errors: List[String]): ProcessState =
     ProcessState(
       deploymentId,
       status,
+      definitionManager.statusName(status),
       version,
       definitionManager.statusActions(status),
       definitionManager.statusIcon(status),
       definitionManager.statusTooltip(status),
+      definitionManager.statusDescription(status),
       startTime,
       attributes,
-      errorMessage
+      errors
     )
 
   @JsonCodec case class StateStatusCodec(clazz: String, value: String)
 }
 
 @JsonCodec case class ProcessState(deploymentId: DeploymentId,
                                    status: StateStatus,
-                                   version: Option[ProcessVersion] = Option.empty,
-                                   allowedActions: List[ProcessActionType] = List.empty,
-                                   icon: Option[URI] = Option.empty,
-                                   tooltip: Option[String] = Option.empty,
-                                   startTime: Option[Long] = Option.empty,
-                                   attributes: Option[Json] = Option.empty,
-                                   errorMessage: Option[String] = Option.empty) {
+                                   name: String,
```

Comment:
can you explain somewhere what is name here? And why it is should be different from status.name?

Reply:
status.name is system stuff, name in state is displayable at FE. If user want do only substitution of nname he can use for example FlinkManager with own implementation of ProcessStateDefinitionManager. In another way he has to implement own statuses and own FlinkManager..